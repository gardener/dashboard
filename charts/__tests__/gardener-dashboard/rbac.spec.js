#!/usr/bin/env node
//
// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

'use strict'

const { basename } = require('path')
const { helm } = fixtures

const chart = basename(__dirname)
const renderTemplates = (templates, values) => helm.renderChartTemplates(chart, templates, values)

describe('gardener-dashboard', function () {
  describe('rbac', function () {
    let templates

    beforeEach(() => {
      templates = [
        'rbac'
      ]
    })

    it('should render the template with default values', async function () {
      const values = {}
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [clusterRoleBinding] = documents
      expect(clusterRoleBinding).toMatchSnapshot()
    })

    it('should not render the template', async function () {
      const values = {
        kubeconfig: 'apiVersion: v1'
      }
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [clusterRoleBinding] = documents
      expect(clusterRoleBinding).toBeFalsy()
    })
  })
})
