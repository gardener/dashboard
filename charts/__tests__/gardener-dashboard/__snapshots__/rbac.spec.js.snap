// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gardener-dashboard rbac should render the template with default values 1`] = `
Object {
  "apiVersion": "rbac.authorization.k8s.io/v1",
  "kind": "ClusterRoleBinding",
  "metadata": Object {
    "labels": Object {
      "app": "gardener-dashboard",
      "chart": "gardener-dashboard-0.2.0",
      "heritage": "Helm",
      "release": "dashboard",
    },
    "name": "garden.sapcloud.io:dashboard:admin",
  },
  "roleRef": Object {
    "apiGroup": "rbac.authorization.k8s.io",
    "kind": "ClusterRole",
    "name": "cluster-admin",
  },
  "subjects": Array [
    Object {
      "kind": "ServiceAccount",
      "name": "gardener-dashboard",
      "namespace": "garden",
    },
  ],
}
`;
